File: frameworks/Java/fair/java-helidon-mp-jpa-data/src/main/java/io/techempower/benchmark/helidon/util/JteUtils.java
package io.techempower.benchmark.helidon.util;

import gg.jte.generated.precompiled.JtefortunesGenerated;
import gg.jte.html.HtmlTemplateOutput;
import gg.jte.html.OwaspHtmlTemplateOutput;
import gg.jte.output.Utf8ByteOutput;
import io.techempower.benchmark.helidon.model.Fortune;

import java.util.List;

public final class JteUtils {

    private JteUtils() { }

    public static byte[] serializeStandard(List<Fortune> fortunes) {
        var out = new Utf8ByteOutput(64);
        HtmlTemplateOutput template = new OwaspHtmlTemplateOutput(out);
        JtefortunesGenerated.render(template, null, fortunes);
        return out.toByteArray();
    }
}

File: frameworks/Java/fair/kotlin-kora2-jdbc-repository/src/main/kotlin/io/techempower/benchmark/kora/util/JteUtils.kt
package io.techempower.benchmark.kora.util

import gg.jte.generated.precompiled.JtefortunesGenerated
import gg.jte.html.OwaspHtmlTemplateOutput
import gg.jte.output.Utf8ByteOutput
import io.techempower.benchmark.kora.model.Fortune

object JteUtils {
    fun serializeStandard(fortunes: List<Fortune>): ByteArray {
        val out = Utf8ByteOutput(256)
        JtefortunesGenerated.render(OwaspHtmlTemplateOutput(out), null, fortunes)
        return out.toByteArray()
    }
}
